pkg/manifests/manifests.go

```diff
@@ -3807,12 +3808,18 @@ func (f *Factory) TelemeterClientPrometheusRule() (*monv1.PrometheusRule, error)
 // TelemeterClientDeployment generates a new Deployment for Telemeter client.
 // If the passed ConfigMap is not empty it mounts the Trusted CA Bundle as a VolumeMount to
 // /etc/pki/ca-trust/extracted/pem/ location.
-func (f *Factory) TelemeterClientDeployment(proxyCABundleCM *v1.ConfigMap) (*appsv1.Deployment, error) {
+func (f *Factory) TelemeterClientDeployment(proxyCABundleCM *v1.ConfigMap, s *v1.Secret) (*appsv1.Deployment, error) {
 	d, err := f.NewDeployment(f.assets.MustNewAssetReader(TelemeterClientDeployment))
 	if err != nil {
 		return nil, err
 	}
 
+	// Set annotation on deplyment to trigger redeployments
+	if s != nil {
+		hash := sha256.New()
+		d.Spec.Template.Annotations["telemeter-token-hash"] = string(hash.Sum(s.Data["token"]))
```

Comment:
For what I see in [the documentation for the flags](https://github.com/openshift/telemeter/blob/master/cmd/telemeter-client/main.go#L59) both the salt and the token are sensitive information, so it would be best not to make any of them public, even obfuscated with a hash function. One idea would be to use some kind of version number for this annotation, and increment it when the token has changed. In `func (f *Factory) TelemeterClientSecret` we already identify the token change in `f.config.ClusterMonitoringConfiguration.TelemeterClientConfig.Token != secretToken`, and we could increment the counter there. One option is storing that number in the secret as `s.Data["revision"]`, which is already available here. That way we are just exposing a counter of how many the token has changed. What do you think?

Reply:
I don't think there is such a huge risk of exposing a secret if it's hashed, specially sha256, however, to try and make it more challenging I've also added the salt to the hash, this way it will make it basically impossible for an attacker to get the secret.

Reply:
This won't expose any sensitive information. Since 'crypto/sha256` notes that "Package sha256 implements the SHA224 and SHA256 hash algorithms as defined in FIPS 180-4". So this code here exposes a cryptographic hash sum of the token. This hash sum is not reversible to the original data by any means currently known.